### PR TITLE
Add flags to link main programs with C/C++ `main`

### PR DESCRIPTION
**Pull request by perazz**
Address #895. I believe this slight change should be enough for now.

### CODE REVIEW

File: src/fpm_targets.f90 [hunk 1]
```diff
@@ -307,6 +307,17 @@ subroutine build_target_list(targets,model)
...
+                    ! If the main program is on a C/C++ source, the Intel Fortran compiler requires option
+                    ! -nofor-main to avoid "duplicate main" errors.
+                    ! https://stackoverflow.com/questions/36221612/p3dfft-compilation-ifort-compiler-error-multiple-definiton-of-main
+                    if (model%compiler%is_intel() .and. any(exe_type==[FPM_TARGET_C_OBJECT,FPM_TARGET_CPP_OBJECT])) then
+                       if (get_os_type()==OS_WINDOWS) then
+                           targets(size(targets))%ptr%compile_flags = '/nofor-main'
+                       else
+                           targets(size(targets))%ptr%compile_flags = '-nofor-main'
+                       end if
+                    end if
```
Comment (awvwgk) [inline src/fpm_targets.f90:319]: Can we put this in the compiler type to provide us with extra link / compile flags?

Comment (perazz) [reply src/fpm_targets.f90:319]: I was trying to do it but unfortunately after that location the link between the `.exe` target and the fact that it comes from a C/C++ source is lost (two separate targets are made: the object and the executable). 

They only need to apply to the linker command when the main program is from a C/C++ source, so, not necessarily general.

Maybe I could add a "main-specific flags" wrapper in the compiler object, to at least hide the Intel-only hack?

**Comment by perazz** [comment]
Please let me know if this is a reasonable solution, so I can go back to working on Intel-MPI on #859

**Comment by perazz** [comment]
Added flags for `pgif90` and `nvfortran`, as suggested in #895

**Review by arteevraina — APPROVED** [review]

**Comment by arteevraina** [comment]
@perazz Looks good to me.

**Comment by perazz** [comment]
Thank you @arteevraina. Any more comments @egiovan? Otherwise, I think this can be merged.

**Review by henilp105 — APPROVED** [review]
Thanks @perazz , Looks good to me.

**Comment by perazz** [comment]
If there are no more comments, I will merge later today.

**Comment by perazz** [comment]
Merging now.

**Pull request merged**